Add weight and length conversion functions

haj8110 opened the pull request.

haj8110 committed:
Add weight and length conversion functions

haj8110 committed:
Add weight and length conversion functions

darwinz reviewed (commented):
Thanks for your contribution!  Please run the linter by running `vendor/bin/phpcs -n`.  There are a few linter errors that need to be addressed.  Thanks!

darwinz commented on Conversions/Lengthconversions.php:
```diff
@@ -0,0 +1,107 @@
+<?php
```
Consider also adding a conversion function for feet to meters and meters to feet

haj8110 replied:
Great idea! I’ll add conversion functions for both feet to meters and meters to feet

darwinz commented on Conversions/Lengthconversions.php:
```diff
@@ -0,0 +1,107 @@
+    private static function validateInput($value, $method) {
+        // Make sure we have a numeric value
+        if (!is_numeric($value)) {
+            throw new InvalidArgumentException("Invalid input for $method");
```
throw new InvalidArgumentException("Invalid input for $method: expected numeric, got " . gettype($value) . " (" . var_export($value, true) . ")");

darwinz replied:
```suggestion
            throw new InvalidArgumentException("Invalid input for $method: expected numeric, got " . gettype($value) . " (" . var_export($value, true) . ")");
```

haj8110 committed:
Update Conversions/Lengthconversions.php

Co-authored-by: Brandon Johnson <bbj1979@gmail.com>

darwinz commented:
@haj8110, please run the linter on your branch and fix the errors.

```
vendor/bin/phpcs -n
```

darwinz commented:
@haj8110 I can't merge until you get the linter to pass.  Are you able to fix the errors?

haj8110 committed:
Refactor LengthConversions and WeightConversions classes for improved code style

- Updated class definitions to follow PSR standards by adding opening braces on the same line.
- Adjusted method definitions to align with coding conventions.
- Ensured consistent formatting and spacing throughout the conversion methods.

haj8110 reviewed (commented):
Update and remove the error
